### ISSUE TITLE
autosectionlabel extension generates warnings

### ISSUE THREAD

**Issue by mgielda** [sphinx-doc/sphinx#2973]
Hi! The autosectionlabel extension is really awesome and we would like to use it extensively. However, it typically generates quite a number of warnings in larger documentations when sections have names like "prerequisites" or "example" several times throughout the document.

In https://github.com/sphinx-doc/sphinx/blob/master/sphinx/ext/autosectionlabel.py#L26 a warning is issued on finding a duplicate label. Wouldn't it be better to issue warnings when ambiguous labels are being used rather than created? In fact it is not relevant that the labels are called the same, unless someone is actually using them - then there is a solution to add another label to the problematic section and the warning would disappear. Is this (warning on use of duplicate level) technically possible and if so, does my proposition make sense (perhaps as a config value)?

Another solution I see is some kind of prefixing, but that is way more extreme.

**Comment by tk0miya** [comment]
This is same behavior as you defined labels::

```
.. _some_label:

.. _some_label:
=> WARNING: Duplicate explicit target name: "some_label".
```

I feel prefixing is much simpler.

**Comment by Skeen** [comment]
Most linting or other checking tools, have ignore statements for this kind of deal.

Simple being able to add a line like:

    .. sphinx.ext.autosectionlabel NOLABEL
    Example
    -------

Would be sufficient for me.

Alternatively merging #3388 would also fix the issue.

**Comment by tk0miya** [comment]
Agreed, #3388 is better solution for this case.
I will take a look later (if I have spare time...)

**Comment by tk0miya** [comment]
Oh, #3388 was already shipped as v1.6.
So this has been resolved.
Thanks,
